pyanaconda/network.py

```diff
@@ -1836,3 +1844,45 @@ def is_ibft_configured_device(iface):
 
 def device_type_is_supported_wired(name):
     return nm.nm_device_type_is_ethernet(name) or nm.nm_device_type_is_infiniband(name)
+
+def set_default_onboot_network(policy, ksdata):
+    """Sets default ONBOOT values and updates ksdata accordingly.
+
+    :param policy: an instance of NetworkOnBoot
+    :param ksdata: a kickstart data
+    """
+    if any(nd.onboot for nd in ksdata.network.network if nd.device):
+        # no need to choose a device
+        return
+
+    if policy is NetworkOnBoot.FIRST_WIRED_WITH_LINK:
+        # choose first wired device having link
+        log.info("Choosing the first wired device having link.")
+
+        for dev in nm.nm_devices():
+
+            if nm.nm_device_type_is_wifi(dev):
+                continue
+            try:
+                link_up = nm.nm_device_carrier(dev)
+            except (nm.UnknownDeviceError, nm.PropertyNotFoundError):
+                continue
+
+            if link_up:
+                update_onboot_value(dev, True, ksdata=ksdata)
+                break
+
+    elif policy is NetworkOnBoot.DEFAULT_ROUTE_DEVICE:
+        # choose the device used during installation
+        # (ie for majority of cases the one having the default route)
+        log.info("Choosing the default route device.")
+
+        dev = default_route_device() or default_route_device(family="inet6")
+        if not dev:
+            return
+
+        # ignore wireless (its ifcfgs would need to be handled differently)
+        if nm.nm_device_type_is_wifi(dev):
+            return
+
+        update_onboot_value(dev, True, ksdata=ksdata)
```

Comment:
Could you please move the `if-elif` content to a separated private methods? I would be easier to read.

Reply:
Fixed.